rust/Dockerfile

```diff
@@ -12,7 +12,7 @@ RUN cargo chef prepare --recipe-path recipe.json
 
 FROM chef as builder
 COPY --from=planner /build/recipe.json .
-RUN cargo chef cook --release --recipe-path recipe.json
+RUN cargo chef cook --all-targets --release --recipe-path recipe.json
```

Comment:
This is my attempt at optimising the build here but I am not actually sure it works. I've noticed that a good chunk is still re-built.

Perhaps we should invoke `cook` three times, one for each package? I also tried to parameterise this here but that seemed to invalidate the layer completely.